src/domain/event/formSections/typeSection/TypeSection.tsx

```diff
@@ -1,18 +1,112 @@
   const { t } = useTranslation();
+  const locale = useLocale();
   const typeOptions = useEventTypeOptions();
 
+  const [{ value: type }] = useField({
+    name: EVENT_FIELDS.TYPE,
+  });
+  const [{ value: hasUmbrella }] = useField({
+    name: EVENT_FIELDS.HAS_UMBRELLA,
+  });
+  const [{ value: isUmbrella }, , { setValue: setIsUmbrella }] = useField({
+    name: EVENT_FIELDS.IS_UMBRELLA,
+  });
+  const [{ value: eventTimes }] = useField<string[]>({
+    name: EVENT_FIELDS.EVENT_TIMES,
+  });
+  const [{ value: recurringEvents }] = useField<string[]>({
+    name: EVENT_FIELDS.RECURRING_EVENTS,
+  });
+
+  const {
+    superEventAtId: savedSuperEvent,
+    superEventType: savedSuperEventType,
+  } = savedEvent
+    ? getEventFields(savedEvent, locale)
+    : { superEventAtId: null, superEventType: null };
+  const superEventSuperEventType = savedEvent?.superEvent?.superEventType;
+  const superEventId = savedEvent?.superEvent?.id;
+
+  const getDisabled = (
+    name: EVENT_FIELDS.HAS_UMBRELLA | EVENT_FIELDS.IS_UMBRELLA
+  ): boolean => {
+    const savedEventIsUmbrellaEvent =
+      savedSuperEventType === SuperEventType.Umbrella;
+    const savedEventIsRecurringEvent =
+      savedSuperEventType === SuperEventType.Umbrella;
+    const savedEventHasSubEvents = Boolean(savedEvent?.subEvents.length);
+    const hasEventTimes = Boolean(eventTimes.length || recurringEvents.length);
+
+    switch (name) {
+      /**
+       * The 'isUmbrella' checkbox should be disabled when:
+       *  - The 'hasUmbrella' checkbox is checked
+       *  - When creating a new event and the form has more than one event date defined for it
+       *  - The event being edited is an umbrella event with sub events
+       *  - The event being edited is a super (recurring) event
+       *  - The event being edited has super event
+       * */
+      case EVENT_FIELDS.IS_UMBRELLA:
+        return (
+          hasUmbrella ||
+          (!savedEvent && hasEventTimes) ||
+          (savedEventIsUmbrellaEvent && savedEventHasSubEvents) ||
+          savedEventIsRecurringEvent ||
+          !!savedSuperEvent
+        );
+      /**
+       * The 'hasHmbrella' checkbox should be disabled when:
+       *  - The 'isUmbrella' checkbox is checked
+       *  - The event being edited is an umbrella event
+       *  - The event being edited is a sub event of a super (recurring) event
+       */
+      case EVENT_FIELDS.HAS_UMBRELLA:
+        return (
+          isUmbrella ||
+          savedEventIsUmbrellaEvent ||
+          superEventSuperEventType === SuperEventType.Recurring
+        );
+    }
+  };
+
+  const disabledIsUmbrella: boolean =
+    hasUmbrella || eventTimes.length || recurringEvents.length;
+
+  useDeepCompareEffect(() => {
+    // Set is umbrella to false if event has more than one event time
+    if ((eventTimes.length || recurringEvents.length) && isUmbrella) {
+      setIsUmbrella(false);
+    }
+  }, [{ eventTimes, isUmbrella, recurringEvents }]);
```

Comment:
Are you sure useDeepCompare is needed here? Woul this works the same way?

```
  React.useEffect(() => {
    // Set is umbrella to false if event has more than one event time
    if ((eventTimes.length || recurringEvents.length) && isUmbrella) {
      setIsUmbrella(false);
    }
  }, [eventTimes, isUmbrella, recurringEvents.length, setIsUmbrella]);
```

Reply:
useDeepCompare might optentially have perf problems. Might not be the case here but I think useDeepCompare shoud be avoided if possbile.

Reply:
Changed as suggested